styles.css

```diff
@@ -1,4 +1,27 @@
-.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .nav-folder-title.file-color-file,
-.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .nav-file-title.file-color-file {
+/* TEXT COLORING, NO CASCADE */
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .file-color-file.file-color-type-text > .nav-folder-title,
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .file-color-file.file-color-type-text > .nav-file-title
+{
   color: var(--file-color-color);
-}
\ No newline at end of file
+}
+
+/* TEXT COLORING, WITH CASCADE */
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .file-color-file.file-color-type-text.file-color-cascade .nav-folder-title,
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .file-color-file.file-color-type-text.file-color-cascade .nav-file-title
+{
+  color: var(--file-color-color);
+}
+
+/* BACKGROUND COLORING, NO CASCADE */
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .file-color-file.file-color-type-background > .nav-folder-title,
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .file-color-file.file-color-type-background > .nav-file-title
+{
+  background-color: var(--file-color-color);
+}
+
+/* BACKGROUND COLORING, WITH CASCADE */
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .file-color-file.file-color-type-background.file-color-cascade .nav-folder-title,
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .file-color-file.file-color-type-background.file-color-cascade .nav-file-title
+{
+  background-color: var(--file-color-color);
+}
```

Comment:
I tested this on a completely fresh install of Obsidian with default themes, no custom CSS and just this plugin installed and it becomes nearly impossible to work with the file explorer as it is for two reasons:

* The current color picker used is just an `<input type="color" />` which doesn't allow for transparency. This means that every element with a colored background covers the vertical lines showing nesting level.
![image](https://github.com/ecustic/obsidian-file-color/assets/2626362/326e281a-e7b4-40ff-b8e7-cf77fd0ac24c)

* Since it is a fixed color switching betwen light and dark theme (which can be controlled by the operating system automatically depending on time of day), causes you to have to swap out all the colors since colors which work in dark mode are rarely legible in light mode and vice versa.
![image](https://github.com/ecustic/obsidian-file-color/assets/2626362/7d6e9352-85ba-487c-b5bc-641c1b38dc5d)

---

So my suggestion for both of the background colors is to force them to be transparent by using `color-mix`:

```suggestion
/* BACKGROUND COLORING, NO CASCADE */
.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .file-color-file.file-color-type-background > .nav-folder-title,
.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .file-color-file.file-color-type-background > .nav-file-title
{
  background-color: color-mix(in srgb, var(--file-color-color) 15%, transparent);
}

/* BACKGROUND COLORING, WITH CASCADE */
.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .file-color-file.file-color-type-background.file-color-cascade .nav-folder-title,
.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .file-color-file.file-color-type-background.file-color-cascade .nav-file-title
{
  background-color: color-mix(in srgb, var(--file-color-color) 15%, transparent);
}
```

Here's how it looks using the exact same colors just swapping the background toggle and light / dark theme. It's far from perfect, but it gives the user a better chance at setting it up to work for them in both themes, and solves the problem of hiding the indentation of the folders.

**No Background - Dark / Light**
![image](https://github.com/ecustic/obsidian-file-color/assets/2626362/5c84582b-4150-4a97-af2b-6c90aea521a7)
![image](https://github.com/ecustic/obsidian-file-color/assets/2626362/aeaa90c3-ce1c-4955-bf4a-8693f517146a)

**Background - Dark / Light**
![image](https://github.com/ecustic/obsidian-file-color/assets/2626362/3905ac1d-83b1-47bd-ac07-94eb3eb3e705)
![image](https://github.com/ecustic/obsidian-file-color/assets/2626362/c9d5a302-a070-4e4d-b3a4-c8233b453dc7)